src/Workspaces/CSharp/Portable/Classification/SyntaxClassification/SyntaxTokenClassifier.cs

```diff
@@ -72,10 +72,11 @@ private static bool CouldBeGenericType(SyntaxToken identifier)
             }
 
             // ?.X.Identifier   or  ?.X.Y.Identifier  is never a generic type.
-            if (identifierName.IsMemberAccessExpressionName() &&
-                identifier.Parent.IsParentKind(SyntaxKind.ConditionalAccessExpression))
+            if (identifierName.IsSimpleMemberAccessExpressionName() ||
+                identifierName.IsMemberBindingExpressionName())
             {
-                return false;
+                if (identifier.Parent.IsParentKind(SyntaxKind.ConditionalAccessExpression))
+                    return false;
```

Comment:
same logic as before, i just didn't want to mix `&&`s and `||`s.  so teh second check moved into a nested check.